### PR TITLE
store/tikv: add pd retry, add tid in commit logs

### LINKED ISSUE #1170: store/tikv: add retry when interacting with PD server

### PR DESCRIPTION
**Pull request by disksing**
Fix #1170 
@shenli @ngaut @siddontang

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by ngaut** [comment]
Please add tests

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -145,26 +145,38 @@ func (c *RegionCache) getRegionFromCache(key []byte) *Region {
...
+			continue
+		}
+		if meta == nil {
+			log.Warnf("region not found for key %q", key)
+			continue
+		}
+		if len(meta.Peers) == 0 {
+			return nil, errors.New("receive Region with no peer")
+		}
+		curPeerIdx := 0
```
Comment (qiuyesuifeng) [inline store/tikv/region_cache.go]: Seems we only use `curPeerIdx = 0` here?

Comment (disksing) [inline store/tikv/region_cache.go]: @qiuyesuifeng The variable has been removed.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by disksing** [comment]
PTAL @ngaut

**Comment by ngaut** [comment]
LGTM

**Pull request merged**